src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/perform_interaction_request.cc

```diff
@@ -346,6 +370,18 @@ bool PerformInteractionRequest::ProcessVRResponse(
     return false;
   }
 
+  if (FirstAnsweredInterface::VR == first_responder_) {
+    // After VR.PerformInteraction response HMI should close UI popup window
+    // if UI.PerformInteraction response comes
+    // after VR.PerformInteraction response.
+    // In this case SDL should send UI_ClosePopUp request.
+    const std::string method_name = "UI.PerformInteraction";
```

Comment:
to avoid code duplication, please add this to a function or repurpose and rename `TerminatePerformInteraction`.

If you create a new function for this, please remove `TerminatePerformInteraction` since it will then be useless.